### PR TITLE
Add eslint-plugin-unicorn

### PR DESCRIPTION
**Pull request by bmish**
All sorts of cool rules for enforcing/converting to better, modern JavaScript code.

Most of this was autofixed. I disabled a few noisy/undesirable rules.

https://github.com/sindresorhus/eslint-plugin-unicorn

### CODE REVIEW

File: lib/rules/no-unused-placeholders.js [hunk 1]
```diff
@@ -57,7 +57,7 @@ module.exports = {
           ) {
             const message = reportInfo.message.value || messageStaticValue.value;
             // https://github.com/eslint/eslint/blob/2874d75ed8decf363006db25aac2d5f8991bd969/lib/linter.js#L986
-            const PLACEHOLDER_MATCHER = /\{\{\s*([^{}]+?)\s*\}\}/g;
+            const PLACEHOLDER_MATCHER = /{{\s*([^{}]+?)\s*}}/g;
```
Comment (aladdin-add) [inline lib/rules/no-unused-placeholders.js:60]: a false negative of the rule [no-useless-escape](https://eslint.org/docs/rules/no-useless-escape)?

File: lib/rules/report-message-format.js [hunk 2]
```diff
@@ -70,7 +70,7 @@ module.exports = {
         messagesObject.value.properties
           .filter(prop => prop.type === 'Property')
           .map(prop => prop.value)
-          .forEach(processMessageNode);
+          .forEach(propValue => processMessageNode(propValue));
```
Comment (aladdin-add) [inline lib/rules/report-message-format.js:73]: maybe just my preference, I like the unchanged code, for more concise. :)

File: .eslintrc.js [hunk 3]
```diff
@@ -7,7 +7,12 @@ module.exports = {
     'not-an-aardvark/node',
     'plugin:node/recommended',
     'plugin:self/all',
+    'plugin:unicorn/recommended',
   ],
+  parserOptions: {
+    ecmaVersion: 2021,
+    sourceType: 'script',
+  },
```
Comment (aladdin-add) [inline .eslintrc.js:15]: this should have been set in `pugin:node/recommended`?

Comment (bmish) [reply .eslintrc.js:15]: eslint-plugin-unicorn sets the `sourceType to `module` so we need to change it back to `script` here.

File: lib/rules/no-unused-placeholders.js (hunk 1 above)
Comment (bmish) [reply lib/rules/no-unused-placeholders.js:60]: Looks like it, [unicorn/better-regex](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/better-regex.md) applies a few more plugins to cleanup the regexp.

File: lib/rules/report-message-format.js (hunk 2 above)
Comment (bmish) [reply lib/rules/report-message-format.js]: Sounds good, I disabled unicorn/no-array-callback-reference](https://github.com/sindresorhus/eslint-plugin-unicorn/blob/main/docs/rules/no-array-callback-reference.md) for now and removed the autofixes.

File: lib/rules/no-unused-placeholders.js (hunk 1 above)
Comment (aladdin-add) [reply lib/rules/no-unused-placeholders.js:60]: can you file the issue in eslint repo, to make sure we don't lost track of it? thanks!

File: .eslintrc.js (hunk 3 above)
Comment (aladdin-add) [reply .eslintrc.js:15]: well, we can remove it when rewrite it in esm. :)

**Review by aladdin-add — APPROVED** [review]
LGTM, thanks! 💯

**Pull request merged**